### PR TITLE
Review ats tg 3.11

### PR DESCRIPTION
**Pull request by PaulaRodrigo-Geograma**
**Redefining the ATS.**

**Tasks executed**

- ✓ Review of the Requirements of the Technical Guide:specifications. 
- ✓ Dependencies with othertests 
- ✓ XPATH expressions 
- ✓ Format of the file on Github 
- ✓ issues closed regarding each test. Modifications discussed to date.

**Most common problems**

- Add/ Remove requirements in each Test.
- Add dependencies for Scenario 2 (Extended Capabilities)
- Xpath redundant expressions
- Wrong * and format
- Delete "input parameter"option of the Xpath expressions
- Two checks that are made are two different tests about WMS and WMTS in the same ATS.

### CODE REVIEW

**Review by michellutz — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -4,10 +4,9 @@ The specification specifies the following conformance classes:
 
 | Conformance class | Standardization target |
 | ----------------- | ---------------------- |
-| [Layer Metadata](http://inspire.ec.europa.eu/id/ats/view-service/3.11/layer-metadata) | OGC Capabilities document |
 | [INSPIRE Profile of ISO 19128 / WMS 1.3.0](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128) | OGC Web Service (WMS 1.3.0) |
-| [INSPIRE Profile of WMTS 1.0.0](http://inspire.ec.europa.eu/id/ats/view-service/3.11/wmts-1.0.0) | OGC Web Service (WMTS 1.0.0) |
 | [INSPIRE Quality of Service](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128-qos) | OGC Web Service (WMS 1.3.0) |
```
Comment (michellutz) [inline README.md]: Shouldn't there be a conformance class also for QoS for WMTS?

File: iso-19128/cascading-view-services.md [hunk 2]
```diff
@@ -10,12 +10,13 @@ This test case only applies for view services that support cascading of other vi
 
 Check that
 * the cascading View Service includes the cascaded services' layer metadata in his own service metadata,
-* the “cascaded” attribute of the <wms:Layer> element is used to indicate that the layer is hosted by a remote View Service and the level of the cascade,
-* the cascading View Service sends requests to the cascaded View Services with the transparency parameter (TRANSPARENT) of the WMS GetMap request set to “true” and the background parameter (BGCOLOR) for all layers  set to the same colour.
+* the 'cascade' attribute of the <wms:Layer> element is used to indicate that the layer is hosted by a remote View Service and the level of the cascade,
```
Comment (michellutz) [inline iso-19128/cascading-view-services.md]: The attribute should be 'cascaded' (with a final 'd')

File: iso-19128/cascading-view-services.md [hunk 3]
```diff
@@ -10,12 +10,13 @@ This test case only applies for view services that support cascading of other vi
...
+* the cascading View Service sends requests to the cascaded View Services with the transparency parameter (TRANSPARENT) of the WMS GetMap request set to 'true' 
+and the background parameter (BGCOLOR) for all layers  set to the same colour.
     
 **Reference(s)**: 
 
-* [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_VS), Requirements 62, 63, 64, 65
```
Comment (michellutz) [inline iso-19128/cascading-view-services.md:18]: Is requirement 62 checked in another test?

File: iso-19128/conformity-node.md [hunk 4]
```diff
@@ -5,15 +5,17 @@
 **Prerequisites**
 
 * [Schema validation](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/schema-validation)
+* [Extended Capabilities](http://inspire.ec.europa.eu/id/ats/view-service/3.11/ISO-19128/extended-capabilities)
 
 **Test method**
 
-This test only applies to [scenario 2](#scenario-2). Otherwise the test case is skipped.
+This test only applies to [Scenario 2](#scenario-2). Otherwise the test case is skipped.
```
Comment (michellutz) [inline iso-19128/conformity-node.md]: How do you check whether scenario 2 is used? It seems in the Xpath expression that you check whether any of the extended capabilities elements are present. Wouldn't it be more appropriate to check whether option 1 is used (and if not, then option 2 is assumed)?

File: iso-19128/contact-position.md [hunk 5]
```diff
@@ -13,7 +13,8 @@
```
Comment (michellutz) [inline iso-19128/contact-position.md:14]: This test only applies to Scenario 2. This should be reflected in the test method (add "This test only applies to [Scenario 2](#scenario-2). Otherwise the test case is skipped.") and XPath expression (add scenario-2), as e.g. in iso-19128/conformity-node.md

File: iso-19128/default-language.md [hunk 6]
```diff
@@ -14,13 +14,11 @@ language can be expected to be used if the capabilities document when no languag
...
 * Execute the request. If the returned resource can be parsed as a valid XML document and if the document passes tests [Schema validation](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/schema-validation) and [supported and response languages node](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/supported-and-response-languages-node):
-  * Check that the [ResponseLanguage code](#response-language) equals the ```lang-code```. If it does pass the test.
-* Otherwise fail the test.
+  * Check that the [ResponseLanguage code](#response-language) equals the ```lang-code```.
 
 **Reference(s)**:
 
-* [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_VS), chapter 4.3.1
-* [TG MD](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_MD), chapter 2.2.7
+* [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_VS), chapter 4.3.1, Requirement 69, 71 , 73
```
Comment (michellutz) [inline iso-19128/default-language.md]: This test only checks requirement 69. Are requirements 71 and 73 checked somewhere else?

File: iso-19128/default-language.md [hunk 7]
```diff
@@ -14,13 +14,11 @@ language can be expected to be used if the capabilities document when no languag
 * Let the [DefaultLanguage code](#default-language) be ```lang-code```.
 * Create a GetCapabilities HTTP request by adding the parameters ```SERVICE=WMS```, ```REQUEST=GetCapabilities```, ```VERSION=1.3.0``` to the ```getcapabilities-url```.
```
Comment (michellutz) [inline iso-19128/default-language.md]: This only checks for a GetCapabilities request without a language. This test should also check for a GetCapabilities request with an unsupported language.

BTW, requirement 69 cannot really be mapped to a requirement in the IR. It should therefore probably be changed into a recommendation.

File: iso-19128/degree-of-conformity.md [hunk 8]
```diff
@@ -5,17 +5,18 @@
...
 This test only applies to [scenario 2](#scenario-2). Otherwise the test case is skipped.
 
-* Check if there is a [Conformity](#Conformity) node in the [ExtendedCapabilities](#ExtendedCapabilities) section. If yes:
+* Check if there is a [Conformity](#Conformity) node in the Extended Capabilities section. If yes:
   * Check that it has a [Deegree](#Degree) node with one of these values: notEvaluated, conformant, notConformant.
 
 **Reference(s)**:
 
-* [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_VS), Chapter 4.2.3.3.1.11
+* [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_VS), Chapter 4.2.3.3.1.11, Requirement 92
```
Comment (michellutz) [inline iso-19128/degree-of-conformity.md]: Should be requirements 22, 23

File: iso-19128/degree-of-conformity.md [hunk 9]
```diff
@@ -5,17 +5,18 @@
 **Prerequisites**
```
Comment (michellutz) [inline iso-19128/degree-of-conformity.md:5]: This test should be merged with / integrated into iso-19128/conformity-node.md

File: iso-19128/fees-node.md [hunk 10]
```diff
@@ -1,3 +1,4 @@
+Capabilities
 # Fees node
```
Comment (michellutz) [inline iso-19128/fees-node.md:2]: This test only applies to Scenario 2. This should be reflected in the test method (add "This test only applies to Scenario 2. Otherwise the test case is skipped.") and XPath expression (add scenario-2), as e.g. in iso-19128/conformity-node.md

File: iso-19128/geographic-bounding-box.md [hunk 11]
```diff
@@ -12,7 +12,7 @@
```
Comment (michellutz) [inline iso-19128/geographic-bounding-box.md:14]: This test only applies to Scenario 2. This should be reflected in the test method (add "This test only applies to Scenario 2. Otherwise the test case is skipped.") and XPath expression (add scenario-2), as e.g. in iso-19128/conformity-node.md

File: iso-19128/harmonised-layer-name.md [hunk 12]
```diff
@@ -1,11 +1,11 @@
 # Harmonised layer name
 
-**Purpose**: It must be unambiguous to find out which of the layers provided by the service visualize the INSPIRE spatial data sets given in the Data Specifications for each INSPIRE theme. These layers must be named according to the INSPIRE Harmonised layer names defined in [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/layer-metadata/README#ref_TG_VS).
+**Purpose**: It must be unambiguous to find out which of the layers provided by the service visualize the INSPIRE spatial data sets given in the Data Specifications for each INSPIRE theme. These layers must be named according to the INSPIRE Harmonised layer names defined in [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/ISO-19128/README#ref_TG_VS).
```
Comment (michellutz) [inline iso-19128/harmonised-layer-name.md:3]: This test needs to be revised. It was originally meant for the WMTS capabilities, which have a different structure than the WMS capabilities. So the test and the XPath references need to be updated accordingly.

The WMTS then needs a corresponding test, that is consistent with the WMTS capabilities structure.

File: iso-19128/inspire-default-styles.md [hunk 13]
```diff
@@ -15,12 +15,12 @@ For each [Layer](#layer) qualified as presenting an inspire harmonized dataset d
 * For each [Style element](#style) within the Layer:
   * If no required default style for this layer is given in the INSPIRE Data Specification in which this harmonized layer name is introduced, pass the test for this layer. Otherwise:
     * Check if the [Style Name](#style-name) equals the name of the default style name for this layer as defined by the Portrayal section of the INSPIRE Data Specification of the INSPIRE theme.
-* If none of the [Style Name](#style-name) elements match, repeat the test recursively for any [Parent Layers](#parent_layer) until a match is found or there is no parent layer (1). If no match was found in the end, fail the test for the original layer.
+* If none of the [Style Name](#style-name) elements match, repeat the test recursively for any Paren Layer until a match is found or there is no parent layer (1). If no match was found in the end, fail the test for the original layer.
```
Comment (michellutz) [inline iso-19128/inspire-default-styles.md]: Why did you remove the link to the XPath?

If kept without a link, it should still read "Parent" (with a final 't')

File: iso-19128/layer-identifier-node.md [hunk 14]
```diff
@@ -8,11 +8,11 @@
 
 **Test method**
 
-* Check if there is a [Identifier](#Identifier) with attribute [@authority](#authority) node in each [Layer](#Layer) section other than layer without [Name](#Name) element formed from daughters
+* Check if there is a [Identifier](#Identifier) with attribute [@authority](#authority) node in each Layer section other than layer without [Name](#Name) element formed from daughters
 * Check whether each [Identifier](#Identifier) node has been declared in an [AuthorityURL](#AuthorityURL) node of the layer itself or its parent layer by heritage by comparing the [@authority](#authority) attribute of the [Identifier](#Identifier) element with the [@name](#AuthorityURLName) attribute of the [AuthorityURL](#AuthorityURL) element.
 
 **Reference(s)**:
-* [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_VS), Chapter 4.2.3.3.4.5
+* [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_VS), Chapter 4.2.3.3.4.5, Requirement 37,38
```
Comment (michellutz) [inline iso-19128/layer-identifier-node.md]: "Requirement 37,38" --> "Requirements 37, 38"

File: iso-19128/in-discovery-service.md [hunk 15]
```diff
@@ -1,24 +0,0 @@
-# View service metadata published in an INSPIRE discovery service
```
Comment (michellutz) [inline iso-19128/in-discovery-service.md:1]: I agree that this test should be removed from the ATS for WMS. However, looking at the requirements 9, 60 and 61, also these should probably be removed or rephrased in the TGs. Please open an issue on the ATS for view services to raise the issue about how to deal with the "Link View Service" operation in the TGs and ATS (and maybe ETS).

File: iso-19128/map-coupled-resource-metadata.md [hunk 16]
```diff
@@ -9,11 +9,11 @@
 **Test method**
 
 * If there is a MetadataURL node for a layer:
```
Comment (michellutz) [inline iso-19128/map-coupled-resource-metadata.md:11]: Not clear (in the TG) whether this applies only to scenario 2. Requirements 13+14 could be intended to implement the requirement in the IR to provide "The Unique Resource Identifier of the resource used to create the layer" as part of the layer metadata. Please open an issue for discussion in the 2017.4 sub-group.

File: iso-19128/map-coupled-resource-metadata.md [hunk 17]
```diff
@@ -9,11 +9,11 @@
...
-  * Verify that the response to a HTTP GET request of the URL in OnlineResource/@xlink:href of the MetadataURL section is an XML document with a root element csw:GetRecordByIdResponse or gmd:MD_Metadata.
-  * In case of a csw:GetRecordByIdResponse document, use the first gmd:MD_Metadata child element. Issue an error if there is no such element.
+  * Verify that the response to a HTTP GET request of the URL in [OnlineResource/@xlink:href](#OnlineResource) of the MetadataURL section is an XML document with a root element csw:GetRecordByIdResponse or gmd:MD_Metadata.
+  * In case of a csw:GetRecordByIdResponse document, use the first [gmd:MD_Metadata](#Metadata) child element. Issue an error if there is no such element.
 
 **Reference(s)**:
-* [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_VS), Chapter 4.2.3.3.1.5
+* [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_VS), Chapter 4.2.3.3.1.5, Requirement 10,13,14
```
Comment (michellutz) [inline iso-19128/map-coupled-resource-metadata.md]: Requirement 10 should be removed here.

File: iso-19128/resource-locator.md [hunk 18]
```diff
@@ -5,15 +5,16 @@
 **Prerequisites**
```
Comment (michellutz) [inline iso-19128/resource-locator.md:5]: We may have to revisit this test, once we have reached an agreement in the 2017.4 sub-group on how to handle the resource locator in the download service (WFS+Atom) - the solution for view services should be the same.

File: iso-19128/review-default-style.md [hunk 19]
```diff
@@ -10,7 +10,7 @@ Review maps for each harmonized layer to verify that the rendering used the spec
 
 **Reference(s)**: 
 
-* [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_VS), Requirements 43, 44.
+* [TG VS](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/README#ref_TG_VS), Chapter 4.2.3.3.4.8, Requirements 43, 44.
```
Comment (michellutz) [inline iso-19128/review-default-style.md]: Thest refers also (mainly) to req 42. Please open an issue in the ATS for discussion in the 2017.4 sub-group whether / how these requirements should be tested for view services.

File: iso-19128/README.md [hunk 20]
```diff
@@ -97,6 +108,7 @@ Based on requirement numbering in [TG VS](#ref_TG_VS).
...
 | 64     | The value of the "cascaded" attribute indicates cascading level | [Review cascading view services](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/cascading-view-services) | |
 | 65     | Transparency & background for collated layers | [Review cascading view services](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/cascading-view-services) | |
+| 66     | Supported languages | [Supported and response languages node](http://inspire.ec.europa.eu/id/ats/view-service/3.11/layer-metadata/supported-and-response-languages-node) | IR Annex III, Part A, Chapter 2.2.3 |
 | 67     | Client may select the language | [Language selection capabilities](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/language-selection-capabilities) | |
 | 68     | GetCapabilities: LANGUAGE parameter | [Language selection capabilities](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/language-selection-capabilities) | |
 | 69     | GetCapabilities: default language | [Default language](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/default-language) | |
```
Comment (michellutz) [inline iso-19128/README.md]: What about req 70-73?

**Review by michellutz — CHANGES_REQUESTED** [review]
See comments.
I did not review the layer-related tests, since I could not see what in detail you changed. It would have be easier to keep them in their original location, make the required changes there (so the reviewers can undestand what exactly has changed) and only then to move them to the new location. But we can always come back to the layers tests if necessary (when we discover issues during the ETS implementation).

File: iso-19128/cascading-view-services.md (hunk 2 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/cascading-view-services.md]: The name of the folder was not correct. The quality of service requirements are common to both: WMS and WMTS --> QoS directory

Comment (PaulaRodrigo-Geograma) [reply iso-19128/cascading-view-services.md]: The attribute name is corrected.

File: iso-19128/cascading-view-services.md (hunk 3 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/cascading-view-services.md:18]: No, requirement 62 is evaluated in this test, where it indicates:
"... the cascading View Service includes the cascaded services' layer metadata in his own service metadata ..."

It was erroneously indicated: the requirement 65 has been exchanged for the 62

File: iso-19128/conformity-node.md (hunk 4 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/conformity-node.md]: It is a good idea to check the situation for scenario 1, but you might not have a section for <inspire_common: MetadataUrl> and even then that does not guarantee that the opposite case (scenario 2) is happening: there may be exceptions when the element <inspire_vs: ExtendedCapabilities> or other errors returned.

File: iso-19128/contact-position.md (hunk 5 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/contact-position.md:14]: Reviewed.

File: iso-19128/map-coupled-resource-metadata.md (hunk 17 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/map-coupled-resource-metadata.md]: Reviewed

File: iso-19128/review-default-style.md (hunk 19 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/review-default-style.md]: https://github.com/inspire-eu-validation/view-service/issues/145

Comment (PaulaRodrigo-Geograma) [reply iso-19128/review-default-style.md]: It is commented on in the issue referred to.

File: iso-19128/README.md (hunk 20 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/README.md]: References to these requirements in the README file have been revised:
(71) default-lenguaje.md
(72) extended-capabilities.md
(73) default-language.md

File: iso-19128/default-language.md (hunk 6 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/default-language.md]: Yes, with the prerequisites: Supported and response languages node

File: iso-19128/degree-of-conformity.md (hunk 8 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/degree-of-conformity.md]: Reviewed.

File: iso-19128/degree-of-conformity.md (hunk 9 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/degree-of-conformity.md:5]: Conformity-node has been merged into degree-of-conformity

File: iso-19128/default-language.md (hunk 7 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/default-language.md]: Reviewed.

File: iso-19128/degree-of-conformity.md (hunk 9 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/degree-of-conformity.md:5]: Reviewed.

File: iso-19128/fees-node.md (hunk 10 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/fees-node.md:2]: Reviewed.

File: iso-19128/geographic-bounding-box.md (hunk 11 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/geographic-bounding-box.md:14]: Reviewed.

File: iso-19128/inspire-default-styles.md (hunk 13 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/inspire-default-styles.md]: Parent Layer is defined as a concept, it applies to any [Layer] element that has a [Layer] group.
I think there is no need for the specific verification of this concept: it does not fall within this requirement to check if there are groupings of [Layer]or more than one "parent" layer.

File: iso-19128/layer-identifier-node.md (hunk 14 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/layer-identifier-node.md]: Reviewed.

**Comment by PaulaRodrigo-Geograma** [comment]
Hello @michellutz Michael,

What is the next step? Should we execute a merge of this pull-request?

I have a question, the content of the review-ats-tg_3.11 branch
(_https://github.com/inspire-eu-validation/view-service/tree/review-ats-tg_3.11_)
should replace the current content in master (not merged). Do you agree?

regards!

File: iso-19128/default-language.md [hunk 21]
```diff
@@ -14,12 +14,12 @@ language can be expected to be used if the capabilities document when no languag
...
 * Execute the request. If the returned resource can be parsed as a valid XML document and if the document passes tests [Schema validation](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/schema-validation) and [supported and response languages node](http://inspire.ec.europa.eu/id/ats/view-service/3.11/iso-19128/supported-and-response-languages-node):
-  * Check that the [ResponseLanguage code](#response-language) equals the ```lang-code```.
+  * Check that all elements gmd:Language equals the ```lang-code```. If it does not, fail the test for this language.
```
Comment (michellutz) [inline iso-19128/default-language.md]: The check should probably be test both the `ResponseLanguage` code and the `gmd:Language` elements.

File: iso-19128/default-language.md (hunk 7 above)
Comment (michellutz) [reply iso-19128/default-language.md]: This still does not check what happens for a request that includes an unsupported language. You should add a second test along these lines:

- Let `unsupported-lang` be a language code that is not included in the supported languages. 
- Create a GetCapabilities HTTP request by adding the parameters ```SERVICE=WMS```, ```REQUEST=GetCapabilities```, ```VERSION=1.3.0``` and `LANGUAGE=unsupported-lang` to the ```getcapabilities-url```.
- Check that ... (See comment below on what should be checked)

File: QoS/README.md [hunk 22]
```diff
@@ -29,4 +29,11 @@ n/a
```
Comment (michellutz) [inline QoS/README.md:31]: The initial part of this conformance class should also include references to WMTS, not just WMS.

**Review by michellutz — CHANGES_REQUESTED** [review]
Some final comments on the default language tests and the intro to the QoS conformance class.

**Comment by michellutz** [comment]
@PaulaRodrigo-Geograma please fix the minor issues. Then the branch will be ready to be merged.

> I have a question, the content of the review-ats-tg_3.11 branch
(https://github.com/inspire-eu-validation/view-service/tree/review-ats-tg_3.11)
should replace the current content in master (not merged). Do you agree? 

The merge will effectively replace the current version of the ATS with the content of the branch.

File: QoS/README.md (hunk 22 above)
Comment (PaulaRodrigo-Geograma) [reply QoS/README.md:31]: 1. The check is added for a request in which a LANGUAGE = unsupported-lang or this empty is specified. The response gets the ResponseLanguage as the DefaultLanguage

2. Reference to web map tiled services is added

File: iso-19128/default-language.md (hunk 21 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/default-language.md]: The check on the ResponseLanguage element code is added

File: iso-19128/harmonised-layer-name.md (hunk 12 above)
Comment (PaulaRodrigo-Geograma) [reply iso-19128/harmonised-layer-name.md:3]: Reviewed.

**Review by michellutz — APPROVED** [review]
